### PR TITLE
🌱 Upgrade Gophercloud to v2

### PR DESCRIPTION
**Pull request by pierreprinetti**
**What this PR does / why we need it**:

This patch upgrades Gophercloud to v2.0.0-rc.3.

This PR exercises CAPO's CI on v2 in preparation for the release.

/hold

### CODE REVIEW

**Comment by pierreprinetti** [comment]
I am not familiar with the test suite here. I suspect there's something wrong with the mocks?

File: pkg/cloud/services/compute/instance.go [hunk 1]
```diff
@@ -99,25 +100,18 @@ func (s *Service) createInstanceImpl(eventObject runtime.Object, instanceSpec *I
...
+		BlockDevice:      blockDevices,
 	}
 
-	blockDevices, err := s.getBlockDevices(eventObject, instanceSpec, instanceSpec.ImageID, instanceCreateTimeout, retryInterval)
-	if err != nil {
-		return nil, err
-	}
-	if len(blockDevices) > 0 {
-		serverCreateOpts = bootfromvolume.CreateOptsExt{
+	server, err := s.getComputeClient().CreateServer(
```
Comment (jichenjc) [inline pkg/cloud/services/compute/instance.go:106]: I am wondering whether you need a the Context.TODO() here in order to make this call happy 
so it's like not related to mock issue,but some param from reall call and test code consistent...

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2107/files#diff-b1233b605e340fb98df3c9fd0a05184538843bc5fb7ee2ce725d944c0868e8b8R267

**Comment by mdbooth** [comment]
This is going to be a merge conflict magnet. Lets aim to get this merged asap when gophercloud v2 is released, or @pierreprinetti is going to be babysitting it almost every day.

**Comment by pierreprinetti** [comment]
>This is going to be a merge conflict magnet

@mdbooth yes indeed. My goals with this PR are:
1. to agree on how to tackle the bump. We already decided that we don't want to wire up context to the whole thing, and that's one decision less to make once it's actually time to merge;
1. to see what CAPO's test machinery thinks;
1. to timidly test whether you folks want to merge a RC instead of waiting for the release :)

File: pkg/cloud/services/compute/instance.go (hunk 1 above)
Comment (pierreprinetti) [reply pkg/cloud/services/compute/instance.go:106]: Matt seems to have found a solution to this issue, I am resolving this conversation. Please let me know if it was not the case!

**Comment by pierreprinetti** [comment]
/hold
Let me just check why Gopherclou2 is not passing CI in cloud-provider-openstack first.

**Comment by EmilienM** [comment]
@pierreprinetti are we good now?

**Comment by pierreprinetti** [comment]
/hold cancel
`v2.0.0` :tada:

**Comment by EmilienM** [comment]
/lgtm
/approve

Thanks for this work!

**Pull request merged**